deprecate the operation.py boolean fns (#7319)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Deprecations 👋</h3>

> * The boolean functions provided by ``pennylane.operation`` are deprecated. See each function's docstring
>   for equivalent code. These include ``not_tape``, ``has_gen``, ``has_grad_method``, ``has_multipar``,
>   ``has_nopar``, ``has_unitary_gen``, ``is_measurement``, ``is_trainable``,
>   ``defines_diagonalizing_gates``, and ``gen_is_multi_term_hamiltonian``.

Reviewer: ```suggestion
* The following boolean functions in ``pennylane.operation`` are now deprecated: :func:`~.operation.not_tape`, 
  :func:`~.operation.has_gen`, :func:`~.operation.has_grad_method`, :func:`~.operation.has_multipar`, :func:`~.operation.has_nopar`, :func:`~.operation.has_unitary_gen`, 
  :func:`~.operation.is_measurement`, :func:`~.operation.`is_trainable`, ``defines_diagonalizing_gates``, and ``gen_is_multi_term_hamiltonian``. 
   Please consult each function's docstring for alternative code. 
   [(#7319)](https://github.com/PennyLaneAI/pennylane/pull/7319)
```

Author: Didn't know we could use sphinx directives in the markdown file 🚀 . Good thing to learn.  Might even be worth circulating this piece of information more widely.

Reviewer 2: Would this also be the fix to current sphinx [complaints](https://productionresultssa6.blob.core.windows.net/actions-results/e18fcf5e-289e-4781-a23f-53dc35f9bb6a/workflow-job-run-5763488f-005a-5e4b-e433-74c6d989d315/logs/job/job-logs.txt?rsct=text%2Fplain&se=2025-04-28T19%3A25%3A51Z&sig=V4G6W%2Bvm3tOdgETE6MICg%2Bqjtuk4vG%2BzGKNvvV%2FBRKs%3D&ske=2025-04-29T06%3A02%3A26Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2025-04-28T18%3A02%3A26Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2025-01-05&sp=r&spr=https&sr=b&st=2025-04-28T19%3A15%3A46Z&sv=2025-01-05)?

<h3>Deprecations 👋</h3>

> * The boolean functions provided by ``pennylane.operation`` are deprecated. See each function's docstring

Reviewer: ```suggestion
* The boolean functions provided by ``pennylane.operation`` are deprecated. See each function's source code 
```

Author: ```suggestion
* The boolean functions provided by ``pennylane.operation`` are deprecated. See the deprecations page
```
I really don't think we should be recommending people look at our source code.

>   for equivalent code. These include :func:`not_tape`, :func:`has_gen`, :func:`has_grad_method`,

Reviewer: ```suggestion
  for equivalent code to use instead. These include :func:`not_tape`, :func:`has_gen`, :func:`has_grad_method`,
```

Author: Merged the suggestion into the other suggestion.

>   for equivalent code. These include :func:`not_tape`, :func:`has_gen`, :func:`has_grad_method`,
>   :func:`has_multipar`, :func:`has_nopar`, :func:`has_unitary_gen`, :func:`is_measurement`,
>   :func:`defines_diagonalizing_gates`, and :func:`gen_is_multi_term_hamiltonian`.

Reviewer: ```suggestion
  for equivalent code. These include :func:`not_tape`, :func:`~.operation.has_gen`, :func:`~.operation.has_grad_method`,
  :func:`~.operation.has_multipar`, :func:`~.operation.has_nopar`, :func:`~.operation.has_unitary_gen`, :func:`~.operation.is_measurement`,
  :func:`~.operation.defines_diagonalizing_gates`, and :func:`~.operation.gen_is_multi_term_hamiltonian`.
```

<h3>Deprecations 👋</h3>

> * The boolean functions provided in `pennylane.operation` are deprecated. See the deprecations page

Reviewer: ```suggestion
* The boolean functions provided in `pennylane.operation` are deprecated. See the :doc:`deprecations page </development/deprecations>` 
```

<h3>Deprecations 👋</h3>

* The boolean functions provided in :mod:`qml.operation <pennylane.operation>` are deprecated. See the deprecations page
  for equivalent code to use instead. These include :func:`~.operation.not_tape`, :func:`~.operation.has_gen`, :func:`~.operation.has_grad_method`,
  :func:`~.operation.has_multipar`, :func:`~.operation.has_nopar`, :func:`~.operation.has_unitary_gen`, :func:`~.operation.is_measurement`,
  :func:`~.operation.defines_diagonalizing_gates`, and :func:`~.operation.gen_is_multi_term_hamiltonian`.
  [(#7319)](https://github.com/PennyLaneAI/pennylane/pull/7319)

> 

Reviewer: Let's just get rid of the directives here (except for qml.operation) and make a link to the deprecations page for code to use instead 👀. I think the deprecations page should be ``:doc:`Deprecations page </development/deprecations>` ``
